Assets/Spawner/ObjectPool.cs

```diff
@@ -34,19 +38,20 @@ public class ObjectPool : MonoBehaviour {
   // Returns an enemy with the given data and position.  If the cooresponding pool is
   // not empty then a pre-created gameObject is returned, otherwise a new one is instantiated.
   public GameObject InstantiateEnemy(EnemyData data, Waypoint start, Transform? parent = null) {
-    if (!objectPools.ContainsKey(data.type)) {
-      prefabMap.Add(data.type, Resources.Load<GameObject>(enemyMap[data.type]));
-      objectPools.Add(data.type, new Queue<GameObject>());
+    EnemyKey enemyKey = new(data.type, data.infectionLevel);
+    if (!objectPools.ContainsKey(enemyKey)) {
```

Comment:
lets make this a function, its used in several places and its getting more involved.